### PR TITLE
Add `--untraced` flag to avoid retesting stories that depend on certain files

### PR DESCRIPTION
**Pull request by ghengeveld**
This introduces a new `--untraced` flag (originally `--ignore-changed`) that can be used in conjunction with `--only-changed` (i.e. TurboSnap). Even if any of the matching files have changed since the baseline build, they will be ignored when tracing dependent stories and so are their dependencies. Dependency tracing simply stops where it encounters a matching file. The flag accepts a glob and can be specified multiple times, just like `--externals`.

### CODE REVIEW

**Comment by ghengeveld** [comment]
@tmeasday This PR does not really address the problem of `index.js` imports. Ignoring that file will not work if it's actually dependencies of that file which have changed. Should we instead stop the graph traversal when we encounter one of these ignored files? WDYT?

**Comment by tmeasday** [comment]
> Should we instead stop the graph traversal when we encounter one of these ignored files? 

That was what I was thinking we should do here, yes.

File: bin-src/tasks/gitInfo.js [hunk 1]
```diff
@@ -114,14 +114,19 @@ export const setGitInfo = async (ctx, task) => {
     try {
       const results = await Promise.all(baselineCommits.map((c) => getChangedFiles(c)));
       ctx.git.changedFiles = [...new Set(results.flat())];
-      ctx.log.debug(`Found changedFiles:\n${ctx.git.changedFiles.map((f) => `  ${f}`).join('\n')}`);
+      ctx.log.debug(
```
Comment (tmeasday) [inline bin-src/tasks/gitInfo.js:117]: I think we should change these logs to not be debug, at least in CI 🤷

**Review by tmeasday — CHANGES_REQUESTED** [review]
>> Should we instead stop the graph traversal when we encounter one of these ignored files?
> That was what I was thinking we should do here, yes.

@ghengeveld are we still planning on addressing this?

File: bin-src/tasks/gitInfo.js (hunk 1 above)
Comment (ghengeveld) [reply bin-src/tasks/gitInfo.js:117]: Done.

**Comment by ghengeveld** [comment]
> > > Should we instead stop the graph traversal when we encounter one of these ignored files?
> > > That was what I was thinking we should do here, yes.
> 
> @ghengeveld are we still planning on addressing this?

Yes, done.

**Review by tmeasday — APPROVED** [review]
LGTM!

**Comment by ghengeveld** [comment]
This was released in 6.3.0.

**Pull request merged**